File: two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> nums, int target) {
        map<int, int> seen;
        for (unsigned int i = 0; i < nums.size(); i++) {
            int number = nums[i];
            int complement = target - number;
            auto it = seen.find(complement);
            if (it != seen.end()) return vector<int>{ it->second, (int)i };
            seen.insert({ number, i });
        }
        return vector<int>{ -1 }; // Unreachable under challenge parameters.
    }
};

File: two-sum.js
function find(arr, target) {
  const lookup = new Map();
  for (let i = 0; i < arr.length; i++) {
    const desired = target - arr[i];
    if (lookup.has(desired)) {
      return [i, lookup.get(desired)];
    } else {
      lookup.set(arr[i], i);
    }
  }
  return [-1, -1];
}
